interface/src/mvd-hosting.d.ts

```diff
@@ -54,9 +55,11 @@ declare namespace MVDHosting {
     spawnApplicationWithTarget(plugin: DesktopPluginDefinition, launchMetadata: any, viewportId: MVDHosting.ViewportId): Promise<MVDHosting.InstanceId>;
     setEmbeddedInstanceInput(embeddedInstance: MVDHosting.EmbeddedInstance, input: string, value: any): void;
     getEmbeddedInstanceOutput(embeddedInstance: MVDHosting.EmbeddedInstance, output: string): Observable<any>|undefined;
-    killApplication(plugin:ZLUX.Plugin, appId:MVDHosting.InstanceId):void;
+    killApplication(plugin:ZLUX.Plugin, appId:MVDHosting.InstanceId,windowID:number):void;
     showApplicationWindow(plugin: DesktopPluginDefinition): void;
+    showApplicationPropertiesWindow(plugin: DesktopPluginDefinition): void;
```

Comment:
The application manager shouldn't directly control the app properties viewer. It should be able to provide app properties to give information to that app properties viewer. But the logic that spawns the app properties viewer should likely be in the UI: for example, with the virtual desktop, each app has an icon, and you can right click the icon to open or close a window of the app. It would be best to request info from the application manager, and spawn the viewer, within that area of the code.

So, doing an action on this line is correct https://github.com/zowe/zlux-app-manager/blob/7ca25862303b591fa68e7ad7047a5876f24e1955/virtual-desktop/src/app/window-manager/mvd-window-manager/launchbar/launchbar/launchbar.component.ts#L147
but the action shouldn't be using the application manager to open the properties window. Instead, try "spawnApplicationWithParms" or "spawnApplication" to spawn the application properties viewer with the info you need.

The difference is that this way, you are not hardcoding which app will be used.